server/tserver/src/main/java/org/apache/accumulo/tserver/log/TabletServerLogger.java

```diff
@@ -348,7 +352,9 @@ public void run() {
               log.info("Our WAL was not used for 12 hours: {}", fileName);
             }
           } catch (InterruptedException e) {
-            // ignore - server is shutting down
+            // Throw an Error, not an Exception, so the AccumuloUncaughtExceptionHandler
+            // will log this then halt the VM.
+            throw new Error("Next log maker thread interrupted", e);
```

Comment:
I'm not sure about this. It seems like it might be a bad idea to use Error for program flow this way. Maybe we should be propagating up the InterruptedException in these cases, and handling it at a higher level to halt? It seems risky to rely on the UEH to deal with this on the assumption that it's going to handle it properly. If there's no other immediate options, I think this is okay... but might be worth keeping a ticket open to pursue a better approach.

Reply:
This is in a background thread. I don't think there is a way to propagate this up.

Reply:
> This is in a background thread. I don't think there is a way to propagate this up.

That's fine if this is the best option today. I haven't spent much time thinking about alternatives, just thinking it'd be useful to have a follow on ticket to try to avoid "new Error" in the code. For this one, I was thinking about what the thread pool executor service would see if the thread was interrupted... that maybe it could be watched like the flusher thread.

I found [this answer](https://stackoverflow.com/a/10038019/196405) about how it should be handled inside a thread pool. The answer seems to assume the thread could only be interrupted by the thread pool, so it's fine to just reset the interrupt flag with Thread.currentThread.interrupt(). I'm not sure if that's always a valid assumption, , but it does seem like it would be difficult for something outside the thread to interrupt it without going through the thread pool.